### ISSUE TITLE
@typescript-eslint/parser' doesn't ignore files, which is **excluded** by/in `tsconfig.json`

### ISSUE THREAD

**Issue by dimabory** [typescript-eslint/typescript-eslint#905]
> The sample project is available here https://github.com/dimabory/ts-eslint-project.

In case of `project` is specified within `parserOptions`, `@typescript-eslint/parser` doesn't ignore files, which is **excluded** by/in `tsconfig.json`.

```json 
// tsconfig
{
  "compilerOptions": {
    "outDir": "./build/",
    "target": "es5",
    "module": "commonjs",
    "esModuleInterop": true
  },
  "include": [
    "src/**/*.ts"
  ],
  "exclude": [
    "src/**/*.test.ts"
  ]
}
```
```js
// .eslintrc.js
module.exports = {
  'env':           {
    'node': true,
    'es6':  true
  },
  'parser':        '@typescript-eslint/parser',
  'extends':       [
    'eslint:recommended',
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended',
  ],
  'parserOptions': {
    'ecmaVersion': 2015,
    'sourceType':  'module',
    'project':     './tsconfig.json'
  },
  'rules':         {}
}

```

**What did you expect to happen?**

**Excluded** files should be ignored by `eslint`.

**What actually happened?**

> _0:0  error  Parsing error: If "parserOptions.project" has been set for @typescript-eslint/parser, .../src/**index.test.ts** must be included in at least one of the projects provided_

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.0.0` |
| `TypeScript`                | `3.5.3` |
| `ESLint`                    | `6.2.2` |
| `node`                      | `10.16.3` |
| `npm`                       | `6.9.0` |

**Comment by dimabory** [comment]
> What command are you running to do your linting?
> Note that includes/excludes from your tsconfig does not dictate what files we report lint errors for - just what files we can parse.

I'm running `eslint src/**/*.ts`. 
I assumed that excluded files in `tsconfig` considered as ignored by the parser as well.  

> If you want to exclude files, either use a `.eslintignore` file, or make sure the globs you pass to the cli is the same as the tsconfig.

The glob is actually the same so that I have to provide additional option in cli to ignore some files `eslint src/**/*.ts --ignore-pattern "*.test.*"`.

**Comment by dimabory** [comment]
Gotcha!
Thank you @bradzacher for the clarity and quick response. 👍

**Comment by TokugawaTakesi** [comment]
@bradzacher , 
I am sorry, but there is noone full match for `tsconfig.eslint.json` in Google search request results for September 2019. Could you please to provide the link to your readme which mentions the `tsconfig.eslint.json`?

**Comment by bradzacher** [comment]
> Could you please to provide the link to your readme which mentions the tsconfig.eslint.json?

Google doesn't always index github readmes fully. You should consider using the github search feature instead.

It's listed in the parser readme:
https://github.com/typescript-eslint/typescript-eslint/blob/4a29098/packages/parser/README.md#L71

As well as the 2.0.0 release notes:
https://github.com/typescript-eslint/typescript-eslint/releases/tag/v2.0.0

There are currently some problems with parsing vue files, because vue is a weird, very custom, use case that is unsupported by typescript. See #883

**Comment by TokugawaTakesi** [comment]
@lvl99 , duplicated tsconfig.json?! Is it normal practice, or just hack?

**Comment by lvl99** [comment]
@TokugawaTakesi it was a quick hack to get working, then I remembered the `extends` option and simplified the file even more. My `tsconfig.eslint.json` now contains minimum 3 options:

```
{
  “extends”: “./tsconfig”,
  “include”: [“src”],
  “exclude”: []
}
```

This lets me target all files in src and exclude none for purposes of compiling TS for eslint.

**Comment by TokugawaTakesi** [comment]
@lvl99, I see, thank you for sharing of your solution!

**Comment by dcDalin** [comment]
> What command are you running to do your linting?
> Note that includes/excludes from your tsconfig does not dictate what files we report lint errors for - just what files we can parse.
> 
> If you want to exclude files, either use a `.eslintignore` file, or make sure the globs you pass to the cli is the same as the tsconfig.

It should be as simple as creating an` .eslintignore` file.

```
node_modules/
public/
bin/
build/
```
